stdlib/public/core/Arrays.swift.gyb

```diff
@@ -1624,23 +1598,22 @@ extension ${Self} {
     return try body(&inoutBufferPointer)
   }
 
-  @discardableResult
   public func _copyContents(
-    initializing ptr: UnsafeMutablePointer<Element>
-  ) -> UnsafeMutablePointer<Element> {
-    if let s = self._baseAddressIfContiguous {
-      let count = self.count
-      ptr.initialize(from: s, count: count)
+    initializing buf: UnsafeMutableBufferPointer<Element>
+  ) -> Iterator? {
+    // FIXME: decide if this really is a fatalError
```

Comment:
"An empty buffer is supposed to be nil." Is it? Surely an empty buffer is just a buffer where `startIndex` == `endIndex` – it can still have a `baseAddress`. i.e. likely cause of empty buffer is that there was no "spare" storage on the end – but that will still have a non-nil pointer inside the buffer, just an zero-sized view onto it. I can understand wanting to sometimes be able to represent an empty buffer by never allocating any memory hence `baseAddress` is optional, but not always, and in this case I suspect that might happen only in error – why would you call `initialize` on a buffer based on memory you haven't allocated?